examples/jsm/loaders/SVGLoader.js

```diff
@@ -1649,6 +1764,14 @@ class SVGLoader extends Loader {
 
 		}
 
+		function isTransformSkewed( m ) {
+
+			const te = m.elements;
+			const basisDot = te[ 0 ] * te[ 3 ] + te[ 1 ] * te[ 4 ];
+			return basisDot !== 0;
+
+		}
```

Comment:
This is testing if the columns are orthogonal. Is this function reliable in practice given there will be round off?

Reply:
It will give false positives, probably only will return false if the matrix is identity. The only solution I think of is testing against a very small angle.

Reply:
In the case of a false-positive it will cause the compute-intensive branch which, nevertheless, gives the correct results for full circles and ellipses.

In practice, for a pure rotation it does not give false-positives since rotation matrices have their conjugate-diagonal set to `sin θ -sin θ`.

The only case to worry about is a long chain of nested group matrix multiplications. If we should handle it, I can update the PR to test against some Epsilon, not zero. Is it a good idea?

Reply:
I think any arbitrary rotation can seem as skewed currently.

It is not ellegant to introduce deltas but I guess the skewing can be detected with a big angle (1 or 2 degrees) as it a very specific use case only for the ellipse.

Reply:
Agree. I’ve added a tolerance a new commit. Please, take a look.

Reply:
Seems right, except for the double division ```basisDot / sx / sy```, it confuses a little, could you put parenthesis?

@WestLangley what do you think about this soluton?

Reply:
I'm inclined to think 1/100 of a degree would be more appropriate. I do not think you are going to get 1 degree round off error.

@Mugen87 What do you think?

Reply:
I would normally start to test against `Number.Epsilon` unless there is a need for a greater threshold. I'm not sure I understand why this is the case for `isTransformSkewed()`.

Reply:
Possible errors can be estimated with some hard-core math and reasoning and I don’t think I’m able to do it. However, yes it sounds that an error should be a magnitude lower than 1°.

I’ve committed a comparison against `Number.EPSILON`. `EPSILON` is of order 1e-16, 1°/100 is of order 1e-8.

Reply:
I just ran this test. It always gets success. With or without random increments, and variations testing against EPSILON or 0, indicates that a matrix constructed with m.rotate() always has ```basisDot``` equal to 0.

So the conclusion is that using Number.EPSILON as threshold works just fine.

```
const m = new Matrix3();
let error = false;
const useRandom = 1;

for ( let a = 0; a < 2.1 * Math.PI; a += 0.00000001 + Math.random() * 0.00000001 * useRandom ) {

    m.identity().rotate( a );

    if ( isTransformSkewed( m ) ) {
        error = true;
        break;
    }

}

console.log( error ? "Error." : "Success." );
```

Reply:
Nice! So, are we ready for the merge?